### ISSUE TITLE
Chatterbox TTS

### ISSUE THREAD

**Issue by subof** [SubtitleEdit/subtitleedit#10792]
Hi,
Along with other technologies, this one seems to have an advantage in accent management. I've tried some others, and they're terribly unmanageable. What's the point of talking about speed or adding tags if the number of supported languages is limited, cross-language transfer is imperfect, and the pronunciation itself sounds incorrect? We're told that fine-tuning is required, but I'm tired of training TTS models and would like to get a working tool out of the box.
Chatterbox TTS seems to sound quite good and is not as resource-intensive.
I would like to ask you to add this advanced technology to the wonderful SE. What could be more fantastic than bringing time measurement back into speech? After all, we live in a world where time plays a crucial role. We have many excellent TTS tools, but the spice of time is rarely available. Thankfully, we have our own Magellan, the author of this program, who navigates his ship through the storms of text. He tautens his sails and firmly holds the helm towards perfect subtitle reading. He has opened a passage, and we have obtained our spice. Now, we need to establish its supply. Chatterbox TTS is probably what many people are looking forward to from this voyage.

https://github.com/resemble-ai/chatterbox

**Comment by darnn** [comment]
It's already being worked on, if I understand correctly:
https://github.com/SubtitleEdit/subtitleedit/pull/10785

**Comment by subof** [comment]
> It's already being worked on, if I understand correctly: [#10785](https://github.com/SubtitleEdit/subtitleedit/pull/10785)

Thanks, I couldn't find this topic in the search.

**Comment by niksedk** [comment]
I was unable to make this work... can you make it work via cmd line?

**Comment by subof** [comment]
> I was unable to make this work... can you make it work via cmd line?

I'm sorry, I don't have any ideas, but I'll pass on this information to you.
AI:
In order for the author of Subtitle Edit (Nikse) to be able to run Chatterbox, he needs to overcome two problems: the lack of a built-in CLI in the library and the difficulties with dependencies in Windows.
Below is a technical comment based on the current working method of using the Turbo version (which is the most stable for integration).
------------------------------
## Hi @Nikse,
Regarding the command line execution, I've put together a more 'verbose' wrapper. It includes a progress indicator for the initial model download (which is ~2.1GB), so the app doesn't look like it's hanging on the first run.
1. Dependencies:

```
pip install torch torchvision torchaudio --index-url https://pytorch.org
pip install chatterbox-tts tqdm
```

2. Python Wrapper (chatter_cli.py):

```
import sys
import torch
import torchaudio
from chatterbox.tts_turbo import ChatterboxTurboTTS
from tqdm import tqdm

def generate_audio(text, output_file):
    device = "cuda" if torch.cuda.is_available() else "cpu"
    print(f"Using device: {device}")
    
    # Chatterbox-turbo is usually the best balance for speed/quality
    print("Loading model (this may take a while on first run)...")
    model = ChatterboxTurboTTS.from_pretrained(device=device)
    
    print(f"Generating audio for: {text[:30]}...")
    wav = model.generate(text)
    
    torchaudio.save(output_file, wav.cpu(), 24000)
    print(f"Done! Saved to {output_file}")

if __name__ == "__main__":
    if len(sys.argv) > 2:
        generate_audio(sys.argv[1], sys.argv[2])
    else:
        print("Usage: python chatter_cli.py \"Text\" \"output.wav\"")
```

Technical Notes for Integration:

* Python Version: Strictly 3.11 is recommended to avoid dependency hell with vector-quantize-pytorch.
* Model Download: The model weights are stored in the Hugging Face cache folder (usually ~/.cache/huggingface). If the download fails, the script will crash with a ConnectionError.
* VRAM: The Turbo model needs about 4GB of VRAM. If the user has an older GPU, the device="cpu" fallback in the script will handle it automatically, though much slower.

This should provide a stable bridge for Subtitle Edit to interface with Chatterbox via Process.Start."

------------------------------
## Why this should work:

   1. Turbo model: I chose this model because it is much faster than the original model and requires less VRAM from the graphics card.
   2. Direct link to Torch: In Windows, a regular pip install torch often installs the CPU version. My command forces the installation of the GPU-enabled version (CUDA 12.1).
   3. Standalone: This script provides a direct path from the text to a file that Nikse can easily call from their C# code.

**Comment by nikse** [comment]
If only I had a dollar each time some tagged me instead of @niksedk

**Comment by subof** [comment]
@niksedk 

Install python3.10: https://www.python.org/ftp/python/3.10.11/python-3.10.11-amd64.exe

Then:
Download the code, extract it, and then navigate to the directory containing `app.py`.   https://github.com/jianchang512/chatterbox-api/archive/refs/heads/main.zip

Navigate to the folder, type "cmd" in the address bar, and press Enter to open the command prompt terminal.

Then execute the following commands in sequence.

```
1. python -m venv venv
2. venv\Scripts\activate
3. pip install -r requirements.txt
4. python app.py
```
If the models don't download, you can get them from here:
https://huggingface.co/ResembleAI/chatterbox/tree/main

I just started it.

**Comment by subof** [comment]
To support CUDA:
1. Activate the environment:
` venv\Scripts\activate`

2. Inside your virtual environment (venv), run the following command:
`pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu121 --force-reinstall`
It is important to use --force-reinstall to replace the default CPU version that was installed automatically.

If you have a new graphics card (RTX 5090 / Blackwell series):
For these cards, it is recommended to use CUDA 12.8 and the corresponding version of PyTorch (e.g., cu128), as older versions do not support their architecture.

How do I check the result? After installation, run the following command:
`python -c "import torch; print(torch.cuda.is_available())"`
If you see True, the message will change to Using device: cuda when you run app.py.

**Comment by niksedk** [comment]
I am waiting for a new release of https://github.com/CrispStrobe/CrispASR/releases (>v0.6.0)
If that will not work, I will try to add support for using the python version.

**Comment by subof** [comment]
> I am waiting for a new release of https://github.com/CrispStrobe/CrispASR/releases (>v0.6.0) If that will not work, I will try to add support for using the python version.

CrispASR doesn't work well, and I wouldn't rely on it anyway. All the TTS projects I've encountered don't use it.

**Comment by subof** [comment]
[in beta-28] Chatterbox TTS doesn't use a GPU, it's a very long time to wait for an answer, and the answer is only in English. For example, he doesn't know German. It also adds extraneous sounds at the beginning and end of generation. There's definitely something wrong, this engine supports many languages and Nvidia GPU.
The samples are completely fine. I tested this on a workstation with an RTX 3090TI.

**Comment by niksedk** [comment]
Chatterbox TTS worked for me using latest CrispASR source code... hopefully he will release an update soon.

**Comment by niksedk** [comment]
Should work in beta 29: https://github.com/SubtitleEdit/subtitleedit/releases
